src/jarabe/journal/listmodel.py

```diff
@@ -92,8 +90,19 @@ def __init__(self, query):
         # avoid hitting D-Bus and disk.
         self.view_is_resizing = False
 
+    def set_query(self, query, clear_selection=True):
```

Comment:
If we were merging this along side #520, we would need to reset the self._updated_entries table here.